### ISSUE TITLE
Spotivy constantly returns "Status Code: 403"

### ISSUE THREAD

**Issue by LingleDev** [danguilherme/spotivy#14]
Hi! I was trying to download a playlist today and I got the error "Error: Status code: 403".
I made sure to run spotivy init and input my credentials (which are freshly generated).
I even generated new ones after that, and those didn't work either.
I really don't know what to do.

**Comment by danguilherme** [comment]
Hello, Thanks for opening the issue!

Could you please post the result of your command with the `--verbose` flag? It helps to check the problem.
Dont forget to omit your credentials when posting it all here.

**Comment by LingleDev** [comment]
Hey! Sorry I haven't responded. My laptop has been out of commission for about a day now. It should be back by Saturday. Thanks for being patient.

**Comment by danguilherme** [comment]
Hey, no problem, good luck with your laptop! :)

**Comment by LingleDev** [comment]
Laptop is up and running now. I can get back to you by tomorrow.

**Comment by danguilherme** [comment]
Hey @FHGDev, any success getting the info?

**Comment by jacobtyq** [comment]
I'm getting similar 403 messages. My youtube API key is active though.
```
[spotivy v0.4.4]
                  [DEBUG] Loaded options:
 {
  "spotifyClientId": "supersecret",
  "spotifyClientSecret": "supersecretclientsecret",
  "youtubeKey": "supersecretyoutubekey",
  "output": "/Users/potato/Repo/spotivy/media",
  "format": "video",
  "quality": "highest",
  "flat": false,
  "audio": false,
  "tracks": [
    "0SFJQqnU0k42NYaLb3qqTx"
  ]
}
Saving media to '/Users/potato/Repo/spotivy/media'

                  [DEBUG] Spotify login
      [Downloading track] Justin Timberlake - Don't Hold the Wall
                  [DEBUG] Search video: search "Justin Timberlake - Don't Hold the Wall"
{ Error: Access Not Configured. YouTube Data API has not been used in project 938759872231 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/youtube.googleapis.com/overview?project=938759872231 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
    at Request._callback (/Users/potato/.nvm/versions/node/v10.16.3/lib/node_modules/spotivy/node_modules/google-auth-library/lib/transporters.js:85:15)
    at Request.self.callback (/Users/potato/.nvm/versions/node/v10.16.3/lib/node_modules/spotivy/node_modules/google-auth-library/node_modules/request/request.js:187:22)
    at Request.emit (events.js:198:13)
    at Request.<anonymous> (/Users/potato/.nvm/versions/node/v10.16.3/lib/node_modules/spotivy/node_modules/google-auth-library/node_modules/request/request.js:1044:10)
    at Request.emit (events.js:198:13)
    at IncomingMessage.<anonymous> (/Users/potato/.nvm/versions/node/v10.16.3/lib/node_modules/spotivy/node_modules/google-auth-library/node_modules/request/request.js:965:12)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  code: 403,
  errors:
   [ { domain: 'usageLimits',
       reason: 'accessNotConfigured',
       message:
        'Access Not Configured. YouTube Data API has not been used in project 938759872231 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/youtube.googleapis.com/overview?project=938759872231 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.',
       extendedHelp:
        'https://console.developers.google.com/apis/api/youtube.googleapis.com/overview?project=938759872231' } ] }
                   [WARN] Video not found!

[spotivy v0.4.4] Finished successfuly
```

**Comment by danguilherme** [comment]
Ok, I was also able to reproduce the error.

It looks like it's a combination of bugs from [Google API Console](https://stackoverflow.com/q/27483674/1574059) and the package used to download YouTube videos, [`ytdl-core`](https://github.com/fent/node-ytdl-core/pull/428).

Here are the steps I followed to fix this:
1. I created a new project in API Console, as suggested in the [StackOverflow thread](https://stackoverflow.com/a/56784302/1574059) (just [disabling/enabling](https://stackoverflow.com/a/27491718/1574059) it didn't work);
2. Enabled [YouTube API](https://console.developers.google.com/apis/library/youtube.googleapis.com?q=Youtube&id=125bab65-cfb6-4f25-9826-4dcc309bc508&project=spotivy-v1&organizationId=0), and create a new key;
3. In my local copy of the spotivy repo, upgraded `ytdl-core` to the latest version (`npm install --save ytdl-core@0.29`);
4. Changed the YouTube key to the new one;
5. Profit!

So on the package side, a new version must be released with the updated `ytdl-core`.

**Comment by danguilherme** [comment]
Fixed in #21

**Comment by LingleDev** [comment]
Thanks for helping us solve this issue. I have not gotten my laptop back up
yet, but I now know what to do when I get it back!

(my laptop went down YET AGAIN)
